docs/release/controllers.md

```diff
@@ -0,0 +1,150 @@
+
+Any changes to the object schema do not require editing or re-creating of Custom Resources.
+Schema fields can't be removed, only new fields can be added with a default value that
+doesn't affect the controller's current behaviour.
+
+A stable API version becomes deprecated once a subsequent stable version is released.
+Stable API versions are not subject to removal in any future release within a controller major version.
+
+In effect, this means that for as long as Flux `v2` is being maintained, all the stable API versions 
+will be supported.
+
+## Controller versioning
+
+The Flux controllers and their Go API packages are released by following the
+[Go module version numbering](https://go.dev/doc/modules/version-numbers) conventions:
+
+- `vX.Y.Z-rc.W` release candidates e.g. `v1.0.0-rc.1`
+- `vX.Y.Z` stable releases e.g. `v1.0.0`
+
+The API versioning and controller versioning are indirectly related. For example,
+a source-controller minor release `v1.1.0` can introduce a new API version
+`v1beta1` for a Kind `XRepository` in the `source.toolkit.fluxcd.io` group.
+
+### Release candidates
+
+Release candidates are intended for testing new features or improvements before a final release.
+
+In most cases, a maintainer will publish a release candidate of a controller for Flux users
+to tests it on their staging clusters. Release candidates are not meant to be deployed in production
+unless advised to do so by a maintainer.
+
+### Patch releases
+
+Patch releases are intended for critical bug fixes to the latest minor version, such as addressing security
+vulnerabilities or fixes to severe problems with no workaround.
+
+Patch releases do not contain breaking changes, feature additions or any type of user-facing changes.
+If a security fix requires a breaking change, then a minor release will provide the fix.
+
+We expect users to be running the latest patch release of a given minor release as soon as the
+controller release is included in a Flux patch release.
+
+### Minor releases
+
+Minor releases are intended for backwards compatible feature additions and improvements.
+Note that breaking changes may occur if required by a security vulnerability fix.
+
+In addition, minor releases are used when updating Kubernetes dependencies such
+as `k8s.io/api` from one minor version to another.
+
+In effect, this means a new minor version will at least be released for all Flux
+controllers approximately every four months, following each Kubernetes minor version release.
+To properly validate the controllers against the latest Kubernetes version,
+we typically allocate a time window of around two weeks for end-to-end testing of Flux controllers.
+
+It is worth noting that in certain scenarios where project dependencies are not in sync with
+the Kubernetes version or conflicts arise, this two-week timeframe may prove insufficient,
+requiring additional time to address the issues appropriately.
```

Comment:
I feel this should probably be moved to the release cadence section, and link to instead. Basically the same approach as I suggested in the `flux.md` document.